### PR TITLE
Handle `wsrep_gtid_mode` enabled

### PR DESCRIPTION
**Pull request by matthieugusmini**
- Changed the parsing logic of the seqno field in the Galera state file
- Added a struct to serialize/deserialize the GTID tied to the sequence number
- Updated the unit tests

Relevant documentation:
- [Using MariaDB GTIDs with MariaDB Galera Cluster](https://mariadb.com/kb/en/using-mariadb-gtids-with-mariadb-galera-cluster/)

### CODE REVIEW

File: pkg/galera/gtid_test.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/mariadb-operator/agent/pkg/galera"
+)
+
+func TestGTIDUnmarshal(t *testing.T) {
+	var gtid galera.GTID
+	err := gtid.UnmarshalText([]byte("0-1`-2"))
```
Comment (mmontes11) [inline pkg/galera/gtid_test.go]:
```suggestion
	err := gtid.UnmarshalText([]byte("0-1-2"))
```
Was the extra quote intended? It's making the tests fail

File: pkg/galera/galera.go [hunk 2]
```diff
@@ -60,25 +61,27 @@ func (g *GaleraState) Compare(other GaleraRecoverer) int {
 	return 0
 }
 
-func (g *GaleraState) Marshal() ([]byte, error) {
+func (g *GaleraState) MarshalText() ([]byte, error) {
```
Comment (mmontes11) [inline pkg/galera/galera.go:64]: Seems like a good idea to implement `TextMarshaler` and `TextUnmarshaler`, more idiomatic

**Review by mmontes11 — CHANGES_REQUESTED** [review]
Thanks for the effort investigating this bug and the implementation of the fix 💯 

Tests are failing, left a suggestion which might probably fix it

File: pkg/galera/gtid_test.go (hunk 1 above)
Comment (matthieugusmini) [reply pkg/galera/gtid_test.go]: Good catch! It was of course not intended aha

**Review by mmontes11 — APPROVED** [review]
LGTM! Thanks for your contribution 🙏🏻 

I will release a `v0.0.4-beta1` to test it together with the operator. Would you mind bumping the agent to this version in the `mariadb-operator` version and raise a PR?

**Pull request merged**